components/feature/pwa/src/main/java/mozilla/components/feature/pwa/WebAppInterceptor.kt

```diff
@@ -0,0 +1,55 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.pwa
+
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import kotlinx.coroutines.runBlocking
+import mozilla.components.concept.engine.EngineSession
+import mozilla.components.concept.engine.request.RequestInterceptor
+import mozilla.components.feature.intent.ext.putUrlOverride
+import mozilla.components.feature.pwa.intent.WebAppIntentProcessor
+
+/**
+ * This feature implements use cases for detecting and handling redirects to standalone web app activities.
+ *
+ * In the case of Android Intents that are not installed, and with no fallback, the user is prompted
```

Comment:
I'm confused, can you explain a bit more about what this interceptor is doing? I don't see any marketplace code here. (although perhaps that's what happens if this class doesn't do any intercepting.)

Could we just move the current Fenix session on install, rather than creating a new interceptor?

Would the interceptor automatically move any requests into the corresponding PWA? For example, if I have the Twitter PWA installed and I happen to open a Twitter link in Fenix, does it automatically open the PWA? Is there a user prompt?

Reply:
Well I wrote the above a little bit late in the day, and I was misreading my own feature request ticket. I can see that this is for intercepting requests and then reopening them in the corresponding installed PWA.

I think we should adjust the KDocs on this class, as the marketplace is mentioned but that's not handled by this interceptor. Theoretically an app could choose to only use this request interceptor. Perhaps something like:
```
This feature will intercept requests and reopen them in the corresponding installed PWA, if any.
```

Reply:
I will fix the KDoc, I just coppied it form the AppLink feature and didn't adjust it properly. 

About your earlier comment:
> Would the interceptor automatically move any requests into the corresponding PWA? For example, if I have the Twitter PWA installed and I happen to open a Twitter link in Fenix, does it automatically open the PWA? Is there a user prompt?
Yes currently it directly will switch into the Twitter PWA. I was also thinking about a possible setting like for app links, or a prompt, but I don't know what's best here UX wise.